peps/pep-0735.rst

```diff
@@ -779,18 +704,47 @@ has impacts on high-level tools like IDEs and Dependabot, which cannot support
 
+Why not support Dependency Group Includes in ``[project.dependencies]`` or ``[project.optional-dependencies]``?
+---------------------------------------------------------------------------------------------------------------
+
+Earlier drafts of this specification allowed Dependency Group Includes to be
+used in the ``[project]`` table.
+However, there were several issues raised during community feedback which led
+to its removal.
+
+Only a small number of additional use cases would be addressed by the inclusion
+of Dependency Groups, and it increased the scope of the specification
+significantly. In particular, this inclusion would increase the number of parties
+impacted by the addition. Many readers of the ``[project]`` table, including build
+backends, SBOM generators, and dependency analyzers are implicated by a change to
+``[project]`` but may continue to operate as-is in the presence of a new (but
+unconnected) ``[dependency-groups]`` table.
+
+Separately from the above concern, the usage modes for
+``[dependency-groups]`` which rely upon inclusion from the ``[project]`` table
+encourage package maintainers to move package metadata from the currently
+standardized location into a newly standardized location. This has a
+negative effect on the simplicity of the mostly static ``pyproject.toml`` metadata
+definition laid out in :pep:`621`.
+
+Finally, exclusion of ``[project]`` support from this PEP is not final. The
+use of includes from that table, or an inclusion syntax from
+``[dependency-groups]`` into ``[project]``, could be introduced by a future
+PEP and considered on its own merits.
+
 Why not support Dependency Group Includes in ``[build-system.requires]``?
 -------------------------------------------------------------------------
 
-Although it may be interesting to allow this in the future, incorporating the
-proposal into the build-system table reduces the ability of users to rely on
-this to bootstrap support. :pep:`517` frontends would be required to support
-Dependency Groups in order to achieve this, and the versions of these frontends
-are not easily controlled by packages.
+Given that we will not allow for ``[project]`` usage of Dependency Groups,
+``[build-system.requires]`` can be considered in comparison with
+``[project.dependencies]``.
 
-By only defining build backend support, it is possible for packages to start
-leveraging the new syntax and capabilities without being concerned about
-controlling the environment in which the package is built and installed.
+There are fewer theoretical usages for build requirements specified in a group
+than package requirements. Additionally, the impact of such a change implicates
+:pep:`517` frontend, which would need to support Dependency Groups in order to
+prepare a build environment. Support in the ``[project]`` table is less
+impactful than this, with stronger motivation, and is still considered out of
+scope.
```

Comment:
I don't quite follow this last sentence.

Reply:
The antecedent for "this" is unclear, so this sentence is probably trying to be too concise.
What I mean is... _tries to word it perfectly_...

> Compared with changes to `[project.dependencies]` and `[project.optional-dependencies]`, changing the behaviors of `[build-system.requires]` is higher impact and has fewer potential uses.
> Therefore, given that this PEP declines to make changes to the `[project]` table, changing `[build-system]` is also rejected.